packages/lib/services/keychain/KeychainServiceDriver.electron.ts

```diff
@@ -0,0 +1,69 @@
+import KeychainServiceDriverBase from './KeychainServiceDriverBase';
+import shim from '../../shim';
+import Logger from '@joplin/utils/Logger';
+import KvStore from '../KvStore';
+import Setting from '../../models/Setting';
+
+const logger = Logger.create('KeychainServiceDriver.node');
+
+const canUseSafeStorage = () => {
+	return !!shim.electronBridge?.()?.safeStorage?.isEncryptionAvailable();
+};
+
+const kvStorePrefix = 'KeychainServiceDriver.secureStore.';
+
+export default class KeychainServiceDriver extends KeychainServiceDriverBase {
+	public override readonly driverId = 'electron-safeStorage';
+
+	public constructor(appId: string, clientId: string) {
+		super(appId, clientId);
+
+		if (canUseSafeStorage() && shim.isLinux()) {
+			logger.info('KeychainService Linux backend: ', shim.electronBridge()?.safeStorage?.getSelectedStorageBackend());
+		}
+	}
+
+	public async supported() {
+		return canUseSafeStorage();
+	}
+
+	public async setPassword(name: string, password: string): Promise<boolean> {
+		if (canUseSafeStorage()) {
```

Comment:
Currently, `canUseSafeStorage` is checked just before each attempt to use `safeStorage`. The goal of this is to handle the case where keychain support becomes unavailable, when previously available. This might not be necessary: `safeStorage` uses the keychain only to store an encryption key. I'm unsure if `canUseSafeStorage` can become `false` when previously `true` without restarting the app.